com.vrcfury.vrcfury/Editor/VF/Service/ActionClipService.cs

```diff
@@ -83,6 +83,29 @@ public class ActionClipService {
                         }
                         break;
                     }
+                    case MaterialPropertyAction materialPropertyAction: {
+                        if (materialPropertyAction.renderer == null && !materialPropertyAction.affectAllMeshes) break;
+                        var renderers = new[] { materialPropertyAction.renderer };
+                        if (materialPropertyAction.affectAllMeshes) {
+                            renderers = avatarObject.GetComponentsInSelfAndChildren<Renderer>();
+                        }
+
+                        foreach (var renderer in renderers) {
+                            var binding = EditorCurveBinding.FloatCurve(
+                                clipBuilder.GetPath(renderer.gameObject),
+                                renderer.GetType(),
+                                $"material.{materialPropertyAction.propertyName}"
+                            );
+                            var defaultSource = renderer.sharedMaterials.Where(mat =>
+                                mat.HasProperty(materialPropertyAction.propertyName)).ToArray();
+                            if (defaultSource.Length != 0) {
+                                offClip.SetConstant(binding, defaultSource[0].GetFloat(materialPropertyAction.propertyName));

```

Comment:
offClip is only needed if the action is forcing something special to happen to the avatar's resting state. In this case (where you're leaving it at the current value), it's not needed so it can be left off (see above in FlipbookAction).

Reply:
Interesting! I'll test to see if removing it works correctly.

I initially made this feature back on version 1.150 and it wasn't picking up resting states from the scene correctly. So as soon as I saw that `offClip` is a thing when I had a moment to make this into a proper PR - I just used it without checking if its actually required.

Will circle back on that when I get a moment and update the PR accordingly

Reply:
I've had issues in the past with material property animations not returning to the resting state (in WD on) (and instead going to the material's default value when not currently animated)
Can't remember if I had this issue with float properties or colour properties

Wonder if that is what you are seeing

Reply:
> I've had issues in the past with material property animations not returning to the resting state (in WD on) (and instead going to the material's default value when not currently animated)

Here specifically, offClip is not actually recorded as a true "clip" in the final avatar -- rather everything put into offClip is just baked into the avatar's resting state. So reading the current value into offClip is effectively a no-op because it just gets put right back into the avatar object and then the clip is thrown away.

If there is a case where unity does not restore a default when WD in on, a fix for that should be in a separate new builder, where we force recording of certain properties that we know unity won't restore properly.

Reply:
Got a chance to look at this, seems to work find without the `offClip` 👍